GitJournal/GitJournal
# Fix bug where saving new note can create duplicate files

## Connection with issue(s)

Resolve issue #638



## Testing and Review Notes

Steps to reproduce/test:
* Tap + to create a new note. The editor appears.
* Type in a title and a note body
* Press android's home button.
* Open android's Recents screen. (Swipe up on newer phones, or the square/rectangle nav bar button on older phones)
* Find GitJournal and tap to open it back up.
* Tap the check mark (top left) to save the note.
* Notice that only 1 copy of the note is saved.

Context / what I learned
* This was happening because the note was already being saved to disk when closing the app https://github.com/GitJournal/GitJournal/blob/9a14048bf67c1cbb2eb20d55eb171fc63f587cf3/lib/editors/note_editor.dart#L242 but then it was getting saved again here (with the new filename) https://github.com/GitJournal/GitJournal/blob/9a14048bf67c1cbb2eb20d55eb171fc63f587cf3/lib/editors/note_editor.dart#L522 The call to `repo.addNote(note)` is basically running `git add .`, and it was picking up both copies of the file.
* I am doing IO operations without adding a try/catch. But the code already is running inside a try/catch with a pretty generic catch block, so any possible errors should be handled gracefully here.



## Screenshots or Videos



![Image 2022-10-16 at 1 09 32 AM](https://user-images.githubusercontent.com/4809268/196020996-48acd42b-e75c-40fa-b8e9-346e204f366d.jpg)


## To Do
